AppVeyor.psm1

```diff
@@ -0,0 +1,364 @@
+        [String]
+        $HarnessFunctionName
+    )
+
+    # Convert the Main Module path into an absolute path if it is relative
+    if (-not ([System.IO.Path]::IsPathRooted($MainModulePath)))
+    {
+        $MainModulePath = Join-Path -Path $env:APPVEYOR_BUILD_FOLDER `
+                                    -ChildPath $MainModulePath
+    }
+
+    $testResultsFile = Join-Path -Path $env:APPVEYOR_BUILD_FOLDER `
+                                 -ChildPath 'TestsResults.xml'
+
+    # Execute custom test task if defined
+    if ($customTaskModuleLoaded `
+        -and (Get-Command -Module $CustomAppVeyorTasks `
+                          -Name Start-CustomAppveyorTestTask `
+                          -ErrorAction SilentlyContinue))
+    {
+        Start-CustomAppveyorTestTask
+    }
+
+    switch ($Type)
+    {
+        'Default'
+        {
+            # Execute the standard tests using Pester.
+            $pesterParameters = @{
+                OutputFormat = 'NUnitXML'
+                OutputFile   = $testResultsFile
+                PassThru     = $True
+            }
+            if ($ExcludeTag.Count -gt 0)
+            {
+                $pesterParameters += @{
+                    ExcludeTag = $ExcludeTag
+                }
+            }
+            if ($CodeCoverage)
+            {
+                Write-Warning -Message 'Code coverage statistics are being calculated. This will slow the start of the tests while the code matrix is built. Please be patient.'
+                $pesterParameters += @{
+                    CodeCoverage = @(
+                        "$env:APPVEYOR_BUILD_FOLDER\*.psm1"
+                        "$env:APPVEYOR_BUILD_FOLDER\DSCResources\**\*.psm1"
+                    )
+                }
+            }
+            $result = Invoke-Pester @pesterParameters
+            break
+        }
+        'Harness'
+        {
+            # Copy the DSCResource.Tests folder into the folder containing the resource PSD1 file.
+            $dscTestsPath = Join-Path -Path $MainModulePath `
+                                      -ChildPath 'DSCResource.Tests'
+            Copy-Item -Path $PSScriptRoot -Destination $MainModulePath -Recurse
+            $testHarnessPath = Join-Path -Path $env:APPVEYOR_BUILD_FOLDER `
+                                         -ChildPath $HarnessModulePath
+
+            # Execute the resource tests as well as the DSCResource.Tests\meta.tests.ps1
+            Import-Module -Name $testHarnessPath
+            $result = & $HarnessFunctionName -TestResultsFile $testResultsFile `
+                                             -DscTestsPath $dscTestsPath
+
+            # Delete the DSCResource.Tests folder because it is not needed
+            Remove-Item -Path $dscTestsPath -Force -Recurse
+            break
+        }
+    }
+
+    $webClient = New-Object -TypeName "System.Net.WebClient"
+    $webClient.UploadFile("https://ci.appveyor.com/api/testresults/nunit/$($env:APPVEYOR_JOB_ID)",
```

Comment:
@TravisEz13 - cool! Thanks! I've added your ```Write-Info``` and ```Push-TestArtifact``` into APpVeyor.psm1 . I also added a few of my own calls to Write-Info so that it was easier to see what was going on. I also added your fix for uploading the test result output.